contracts/libraries/PositionValue.sol

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity >=0.7.0;
+
+import '@uniswap/v3-core/contracts/interfaces/IUniswapV3Pool.sol';
+import '@uniswap/v3-core/contracts/libraries/FixedPoint128.sol';
+import '@uniswap/v3-core/contracts/libraries/TickMath.sol';
+import '../interfaces/INonfungiblePositionManager.sol';
+import './LiquidityAmounts.sol';
+import './PoolAddress.sol';
+import './PositionKey.sol';
+
+library PositionValue {
+    function principal(INonfungiblePositionManager nft, uint256 tokenId)
+        internal
+        view
+        returns (uint256 amount0, uint256 amount1)
+    {
+        (, , address token0, address token1, uint24 fee, int24 tickLower, int24 tickUpper, uint128 liquidity, , , , ) =
+            nft.positions(tokenId);
```

Comment:
what about tokens owed to the position but not collected returned by this method? include in fees or principal? i think fees because they are not subject to IL